text/0000-build-std/7-stage-3.md

```diff
@@ -0,0 +1,220 @@
+Configurations like "match-profile" for `build-std` make most sense when
+combined with Cargo profiles that aim to maximise the optimisation of the final
+binary. It is more likely that users would want to use "match-profile" with the
+release profile than by default (as in `[build]`) or for a specific target (as
+in `[target]`).
+
+↩ [*Proposal*][proposal]
+
+## Why does `[profile]` have higher precedence than `[build]` and lower than `[target]`?
+[rationale-profile-precedence]: #why-does-profile-have-higher-precedence-than-build-and-lower-than-target
+
+`[target]` configuration is more narrowly scoped than `[profile]` which is in
+turn more narrowly scoped than the global default in `[build]`.
+
+↩ [*Proposal*][proposal]
+
+## Why have "match-profile" as the default for the release profile?
+[rationale-default]: #why-have-match-profile-as-the-default-for-the-release-profile
+
+`build-std = "match-profile"` is intended to be used when additional time
+spent building the standard library is not a problem and the quality of the
+final artifact is paramount. This corresponds closely with the release profile,
+where additional time spent on optimisations (e.g. with `-Ctarget-cpu`) is
+acceptable. Always re-building the standard library with the user's profile
+configuration in release mode is likely to result in a more optimised build than
+with the pre-built standard library and is thus a reasonable default for the
+`release` and `bench` profiles.
+
+↩ [*Proposal*][proposal]
+
+### Why not always default to "match-profile"?
+[rationale-why-not-always-rebuild]: #why-not-always-default-to-match-profile
+
+Cargo's users don't currently expect that changing any part of their profile
+configuration, such as trying a different optimisation level, would trigger a
+rebuild of the standard library. For small projects, rebuilding the standard
+library could be a significant increase in the overall build time for a project.
+
+If `build-std = "match-profile"` were the default, the standard library
+could be rebuilt quite frequently without much benefit. Especially as the
+pre-built standard library is built using the release profile, all debug profile
+builds would immediately trigger a rebuild of the standard library.
+
+↩ [*Proposal*][proposal]
+
+### Why add "compatible-profile"?
+[rationale-compatible-profile]: #why-add-compatible-profile 
+
+"compatible-profile" is useful for when users want a more debuggable standard
+library while keeping rebuilds of the standard library to a minimum.
+
+↩ [*Proposal*][proposal]
+
+### Why does "compatible-profile" use the standard library's profiles?
+[rationale-compatible-profile-std]: #why-does-compatible-profile-use-the-standard-librarys-profiles 
+
+By using the standard library's profile definitions, the library team will be
+able to define a "dev" profile that is most useful for the standard library.
+
+↩ [*Proposal*][proposal]
+
+### Why add "match-profile"?
+[rationale-match-profile]: #why-add-match-profile 
```

Comment:
Note that a more general solution will be needed for opaque dependencies *but* designing a general solution now without the rest of the feature is not ideal and this seems narrowly scoped enough to not conflict with anything we do there.